core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -182,10 +183,17 @@ protected SqlImplementor(SqlDialect dialect) {
   /** Visits a relational expression that has no parent. */
   public final Result visitRoot(RelNode r) {
     RelNode best;
-    if (!this.dialect.supportsGroupByLiteral()) {
+    if (!this.dialect.supportsGroupByLiteral()
```

Comment:
I originally planned to directly convert the Join into an SqlNode in the RelToSqlConverter. This approach seemed quite reasonable now.